android-lib/src/main/java/com/kinvey/android/cache/RealmCache.java

```diff
@@ -282,9 +282,46 @@ public int delete(Query query) {
                 QueryHelper.prepareRealmQuery(realmQuery, query.getQueryFilterMap());
 
                 RealmResults result = realmQuery.findAll();
+
                 ret = result.size();
-                result.deleteAllFromRealm();
-                mRealm.commitTransaction();
+                int limit = query.getLimit();
+                int skip = query.getSkip();
+
+                if (limit > 0)
+                {
+                    // limit modifier has been applied, so take a subset of the Realm result set
+                    int endIndex = Math.min(ret, (skip+limit));
+                    List<DynamicRealmObject> subresult = result.subList(skip, endIndex);
+                    List<String> ids = new ArrayList<String>();
+                    ret = subresult.size();
+                    for (DynamicRealmObject id : subresult) {
+                        ids.add((String)id.get("_id"));
+                    }
+                    mRealm.commitTransaction();
+                    if (ids.size() > 0) {
+                        this.delete(ids);
+                    }
+                } else if (skip > 0) {
+                    // only skip modifier has been applied, so take a subset of the Realm result set
+                    if (skip < result.size()) {
+                        List<DynamicRealmObject> subresult = result.subList(skip, result.size());
+                        List<String> ids = new ArrayList<String>();
+                        ret = subresult.size();
+                        for (DynamicRealmObject id : subresult) {
+                            ids.add((String) id.get("_id"));
+                        }
+                        mRealm.commitTransaction();
+                        this.delete(ids);
+                    }
+                    else {
+                        ret = 0;
+                    }
+                } else {
+                    // no skip or limit applied to query, so delete all results from Realm
+                    result.deleteAllFromRealm();
+                    mRealm.commitTransaction();
+                }
+
             } else {
```

Comment:
If here will be false? If query will be contained `in` operator? 
`if (!isQueryContainsInOperator(query.getQueryFilterMap()))`